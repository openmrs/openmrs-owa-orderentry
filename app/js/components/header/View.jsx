import React from 'react';
import PropTypes from 'prop-types';

const contextPath = window.location.href.split('/')[3];

const View = props => (
  <header>
    <div className="logo">
      <a href="../../referenceapplication/home.page">
        <img src="../../ms/uiframework/resource/uicommons/images/logo/openmrs-with-title-small.png" alt="" />
      </a>
    </div>

    <ul className="user-options">
      <li
        role="button"
        tabIndex="0"
        className="identifier"
        style={{ cursor: "pointer" }}
        onClick={() => { props.toggleState("userDropdown"); }}
      >
        <i className="icon-user small" />
        {props.currentUser}

        <i className="icon-caret-down appui-icon-caret-down link" /><i className="icon-caret-up link appui-toggle" style={{ display: "none" }} />
        {props.userDropdown &&
          <ul id="user-account-menu">
            <li>
              <a
                href="../../adminui/myaccount/myAccount.page">Account
              </a>
            </li>
          </ul>
        }

      </li>
      <li className="change-location">
        <a href="#" onClick={() => { props.toggleState("locationDropdown"); }}>
          <i className="icon-map-marker small" />
          <span id="selected-location">{props.currentLocation.display}</span>

          <i className="link icon-caret-down" />
        </a>
        {props.locationDropdown &&
          <div>
            <ul className="location-container">
              {props.locations.map(location => (
                <li
                  role="button"
                  key={location.uuid}
                  onClick={() => {
                    props.setCurrentLocation(location.uuid);
                    props.toggleState("locationDropdown", false);
                  }}
                  className={location.display === props.currentLocation.display ? "selected" : ""}
                >
                  {location.display}
                </li>
              ))}
            </ul>
          </div>
        }
      </li>
      <li className="logout">
        <a href={`../../appui/header/logout.action?successUrl=${contextPath}`}>
          Logout
          <i className="icon-signout small" />
        </a>
      </li>
    </ul>
  </header>
);

View.propTypes = {
  currentLocation: PropTypes.shape().isRequired,
  currentUser: PropTypes.string.isRequired,
  locations: PropTypes.array.isRequired,
};

export default View;
