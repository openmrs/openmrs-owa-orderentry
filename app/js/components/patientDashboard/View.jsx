import React from 'react';
import dateFns, { distanceInWordsToNow } from 'date-fns';
import PropTypes from 'prop-types';

import '../../../css/patient.css';

const View = props => (
  <div className="patient-header ">
    <div className="demographics">
      <h1 className="name">
        <span>
          <span className="PersonName-givenName">{props.patient.person.personName.givenName}&nbsp;&nbsp;</span>
          <em>Given</em>
        </span>

        <span>
          <span className="PersonName-middleName">{props.patient.person.personName.middleName}&nbsp;&nbsp;</span>
          <em>Middle</em>
        </span>

        <span>
          <span className="PersonName-familyName">{props.patient.person.personName.familyName}</span>
          <em>Family Name</em>
        </span>

        &nbsp;
        <span className="gender-age">
          <span>{props.patient.person.gender === 'M' ? "Male" : "Female"}&nbsp;</span>
          <span>
            {props.patient.person.age} year(s) ({dateFns.format(new Date(props.patient.person.birthdate), 'DD[.]MMM[.]YYYY')})
          </span>
          <span id="edit-patient-demographics" className="edit-info">
            <small>
              <a href="../../registrationapp/editSection.page?patientId=107&sectionId=demographics&appId=referenceapplication.registrationapp.registerPatient">Edit</a>
            </small>
          </span>
          <a
            role="button"
            tabIndex="0"
            onClick={() => { props.toggleDetailsView(); }}
            id="patient-header-contactInfo" className="contact-info-label expanded"
          >
            {props.showContactInfo ?
              <span>Hide Contact Info <i className="toggle-icon icon-caret-up small" /></span> :
              <span>Show contact info <i className="toggle-icon icon-caret-up small rotate180" /></span>
            }
          </a>
        </span>

        <div className="firstLineFragments" />

        {props.showContactInfo && props.patient.attributes &&
          <div className="" id="contactInfoContent">
            <div className="contact-info-inline">
              <span>
                {props.patient.person.preferredAddress.display}
                <em>Address</em>
              </span>
              <span className="left-margin">
                <span id="coreapps-telephoneNumber">
                  {props.patient.attributes[0].value}
                </span>
                <em>Telephone Number</em>
              </span>
              &nbsp;&nbsp;
              <small id="contact-info-inline-edit" className="edit-info">
                <a href="../../registrationapp/editSection.page?patientId=107&sectionId=contactInfo&appId=referenceapplication.registrationapp.registerPatient">Edit</a>
              </small>
            </div>
          </div>
        }
      </h1>
    </div>

    <div className="identifiers">
      <em>Patient ID</em>
      <span>{props.patient.patientIdentifier.identifier}</span>
      <br />
    </div>

    {props.note.length > 0 &&
      <div className="secondLineFragments">
        <div className="clickToEditObs ng-scope">
          <div className="firstLine">
            <span className="note-wrapper">
              <span>
                <pre className="preformatted-note">{props.note[0].value}</pre>
              </span>
            </span>
          </div>
          <div className="details secondLine">
            <span className="created-by">{props.note[0].auditInfo.creator.display}
            </span>&nbsp;
            <span className="created-date">
              {distanceInWordsToNow(props.note[0].auditInfo.dateCreated, {
                includeSeconds: true, addSuffix: true,
              })}
            </span>
          </div>
        </div>
      </div>
    }
  </div>
);

View.propTypes = {
  note: PropTypes.array,
  patient: PropTypes.shape().isRequired,
  showContactInfo: PropTypes.bool.isRequired,
  toggleDetailsView: PropTypes.func.isRequired,
};

View.defaultProps = {
  note: [],
};

export default View;
