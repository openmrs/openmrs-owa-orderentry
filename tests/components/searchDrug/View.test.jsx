import React from 'react';
import View from '../../../app/js/components/searchDrug/View';

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<View {...props} />);
  }
  return mountedComponent;
};

describe('Component: searchDrug: View', () => {
  beforeEach(() => {
    props = {
      results: [
        { display: 'paracetamol', uuid: '1232h-2323' },
        { display: 'paraphenol', uuid: '8932h-2323' }
      ],
      loading: false,
      searchDrug: jest.fn(),
    };
    mountedComponent = undefined;
  });

  describe('Conditional rendering', () => {
    it('renders properly', () => {
      const component = getComponent();
      expect(component).toMatchSnapshot();
    });

    it('show Searching...', () => {
      props.loading = true;
      const component = getComponent().find('span').at(1);

      expect(component.props().children).toBe(' Searching...');
    });

    it('show search error...', () => {
      props.error = { data: { message: 'Network error' } };
      const component = getComponent().find('span').at(2);

      expect(component.props().children).toBe('Network error');
    });
  });

  describe('Search drug action', () => {
    it('calls setCurrentLocation()', () => {
      const event = { target: { value: 'para' } };
      const searchDrugSpy = jest.spyOn(props, 'searchDrug');
      const drugSearchInput = getComponent().find('p input').first();
      drugSearchInput.simulate('change', event);

      expect(searchDrugSpy).toHaveBeenCalledTimes(1);
    });
  });
});
