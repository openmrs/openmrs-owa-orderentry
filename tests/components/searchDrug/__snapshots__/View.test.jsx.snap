// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: searchDrug: View Conditional rendering renders properly 1`] = `
<div
  className="search-container"
>
  <p>
    New order for: 
  </p>
  <span>
    <input
      id="sel-drug4841input"
      onChange={[Function]}
      placeholder="Search for drug...(3 chars min.)"
      type="text"
      value=""
    />
  </span>
</div>
`;
